agent/src/main/java/com/teamscale/jacoco/agent/util/ClasspathUtils.java

```diff
@@ -0,0 +1,35 @@
+package com.teamscale.jacoco.agent.util;
+
+import com.teamscale.jacoco.agent.options.AgentOptionParseException;
+import com.teamscale.jacoco.agent.options.FilePatternResolver;
+import org.conqat.lib.commons.collections.CollectionUtils;
+import org.conqat.lib.commons.filesystem.FileSystemUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/** Handles parsing a .txt file with a classpath. */
+public class ClasspathUtils {
+	/** Replaces all txt files in the given list with the file names written in the txt file separated by new lines. */
+	public static List<File> resolveClasspathTextFiles(String key, FilePatternResolver filePatternResolver,
+													   List<String> classDirectoriesOrZips) throws AgentOptionParseException, IOException {
+		List<File> classDirectoriesOrZipFiles = CollectionUtils
+				.mapWithException(classDirectoriesOrZips, file -> filePatternResolver.parseFile(key, file));
```

Comment:
If I understand correctly, this will resolve each pattern in the file to a *single* file. However, I think in this case it makes more sense to allow resolution to any number of files. E.g. so I can specify `/foo/bar/**/*.jar`

Reply:
Had to do some more refactorings in FilePatternResolver, but is done now